contrib_bots/lib/github_issues.py

```diff
@@ -0,0 +1,104 @@
+from . import github
+import os
+import requests
+import urllib.request, urllib.error, urllib.parse
+
+class IssueHandler(object):
+    '''
+    This plugin facilitates sending issues to github, when
+    an item is prefixed with '@issue'
+    It will also write items to the issues stream, as well
+    as reporting it to github
+    '''
+    CHARACTER_LIMIT = 70
+
+    def usage(self):
+        return '''
+            This plugin will allow users to flag messages
+            as being issues with Zulip by using te prefix '@issue'
+            Before running this, make sure to create a stream
+            called "issues" that your API user can send to.
+            Also, make sure that the credentials of the github bot have
+            been typed in correctly, that there is a personal access token
+            with access to public repositories ONLY,
+            and that the repository name is entered correctly.
+            Check ~/.github/github.conf, and make sure there are
+            github_repo   (The name of the repo to post to)
+            github_repo_owner   (The owner of the repo to post to)
+            github_username   (The username of the github bot)
+            github_token    (The personal access token for the github bot)
+            '''
+
+    def triage_message(self, message):
+        # return True if we want to (possibly) respond to this message
+        original_content = message['content']
+        # This next line of code is defensive, as we
+        # never want to get into an infinite loop of posting follow
+        # ups for own follow ups!
+        if message['display_recipient'] == 'issue':
+            return False
+        is_issue = original_content.startswith('@issue')
+        return is_issue
+
+    def handle_message(self, message, client, state_handler):
+
+        original_content = message['content']
+        original_sender = message['sender_email']
+
+        new_content = original_content.replace('@issue', 'by {}:'.format(original_sender,))
+
+        gh = github.Github()
+        # signs into github using the provided username and password
+        gh.auth()
+
+        # Gets rid of the @issue in the issue title
+        issue_title = message['content'].replace('@issue', '').strip()
+        issue_content = ''
+        new_issue_title = ''
+        for part_of_title in issue_title.split():
+            if len(new_issue_title) < self.CHARACTER_LIMIT:
+                new_issue_title = ' '.join([new_issue_title, part_of_title])
```

Comment:
I think you can just use `new_issue_title += part_of_title` here, even though this is an inefficient approach since strings are immutable in Python. I think creating an efficient solution will complicate your code.